### ISSUE TITLE
Events chart crashing

### ISSUE THREAD

**Issue by vargaslucas** [umami-software/umami#2498]
### Describe the Bug

**Context**
I noticed this bug on the website dashboard happening when a new event track is triggered from a landing page, either by data attributes or custom js trigger, causing the `datasets` object to update, and due to the code below, we are trying to update `chart.data.datasets` with the updated `datasets` while the 2 object lengths don't match.

`src/components/metrics/BarChart.js:125`
```
datasets.forEach((dataset, index) => {
  chart.current.data.datasets[index].data = dataset.data;
  chart.current.data.datasets[index].label = dataset.label;
});
```

**Steps to Reproduce**
- create new website
- add website-id to html page with the tracking script
- in umami, open up the new website, go to any section details by clicking `More ->`
- on html page trigger an event using data-attributes or the umami object
- on umami, click `<- Back` button

**Solution Proposal**
I did "fixed" the issue on my end with this code snippet, replacing the loop with a simple object rewrite.
```
chart.current.data.datasets = datasets;
```

**Questions**
1. Was there a specific reason for the current implementation using a loop to update the chart dataset individually?
2. Should I open a Pull Request to address the issue?

### Database

MySQL

### Relevant log output

```shell
7864-28222780cf9b3273.js:1 TypeError: Cannot set properties of undefined (setting 'data')
    at 4263-d577d661349e5a61.js:1:8851
    at Array.forEach (<anonymous>)
    at E (4263-d577d661349e5a61.js:1:8804)
    at 4263-d577d661349e5a61.js:1:9007
    at ap (fd9d1056-dcf638336f76c32a.js:9:68405)
    at aM (fd9d1056-dcf638336f76c32a.js:9:79901)
    at aO (fd9d1056-dcf638336f76c32a.js:9:80539)
    at aR (fd9d1056-dcf638336f76c32a.js:9:80423)
    at aO (fd9d1056-dcf638336f76c32a.js:9:80519)
    at aR (fd9d1056-dcf638336f76c32a.js:9:80423)
```

**Comment by Maxime-J** [comment]
I saw that error a few times while testing, thinking it was due to some bug after console usage, but obviously not!

Nice you pinpointed the cause 👍 
however, always reassigning datasets with the new ones causes a not so pretty visual chart refresh when switching pages.
It would likely be better to keep the existing loop when datasets are of same length, and do your fix otherwise.

PR are taken into account by the team, but if you don't want or don't have the time to submit one,
this will get fixed anyway :)
